src/vs/editor/contrib/inlineEdit/browser/inlineEditController.ts

```diff
@@ -0,0 +1,311 @@
+		return editor.getContribution<InlineEditController>(InlineEditController.ID);
+	}
+
+	private _currentEdit: ISettableObservable<InlineEditWidget | undefined> = disposableObservableValue(this, undefined);
+	private _currentRequestCts: CancellationTokenSource | undefined;
+
+	private _jumpBackPosition: Position | undefined;
+	private _isAccepting: boolean = false;
+
+	private readonly _enabled = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).enabled);
+	private readonly _fontFamily = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).fontFamily);
+	private readonly _backgroundColoring = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).backgroundColoring);
+
+
+	constructor(
+		public readonly editor: ICodeEditor,
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+		@ILanguageFeaturesService private readonly languageFeaturesService: ILanguageFeaturesService,
+		@ICommandService private readonly _commandService: ICommandService,
+	) {
+		super();
+
+		//Automatically request inline edit when the content was changed
+		//Cancel the previous request if there is one
+		//Remove the previous ghost text
+		this._register(editor.onDidChangeModelContent(async () => {
+			if (!this._enabled.get()) {
+				return;
+			}
+			await this.getInlineEdit(editor, true);
+		}));
+
+		//Check if the cursor is at the ghost text
+		this._register(editor.onDidChangeCursorPosition((e) => {
+			if (!this._enabled.get()) {
+				return;
+			}
+			this.checkCursorPosition(e.position);
+		}));
+
+		//Perform stuff when the current edit has changed
+		this._register(autorun((reader) => {
+			/** @description InlineEditController.update model */
+			const currentEdit = this._currentEdit.read(reader);
+			if (!currentEdit) {
+				this._isVisibleContext.set(false);
+				this._isCursorAtInlineEditContext.set(false);
+				return;
+			}
+			this._isVisibleContext.set(true);
+			this._isCursorAtInlineEditContext.set(false);
+			const pos = editor.getPosition();
+			if (pos) {
+				this.checkCursorPosition(pos);
+			}
+		}));
+
+		const styleElement = this._register(createStyleSheet2());
+		this._register(autorun(reader => {
+			const fontFamily = this._fontFamily.read(reader);
+			styleElement.setStyle(fontFamily === '' || fontFamily === 'default' ? `` : `
+.monaco-editor .inline-edit-decoration,
+.monaco-editor .inline-edit-decoration-preview,
+.monaco-editor .inline-edit {
+	font-family: ${fontFamily};
+}`);
+		}));
+
+		const backgroundRemoved = this._register(createStyleSheet2());
+		this._register(autorun(reader => {
+			const backgroundColoring = this._backgroundColoring.read(reader);
+			backgroundRemoved.dispose();
+			backgroundRemoved.setStyle(backgroundColoring ? `
```

Comment:
Optional: Can you solve this via a class (`.inline-edit-remove.backgroundColoring`)? Generally, we try to avoid dynamic JS CSS manipulation.